components/pluralrules/src/lib.rs

```diff
@@ -281,29 +272,41 @@ impl PluralRules {
     /// ```
     /// use icu_locale::LanguageIdentifier;
     /// use icu_pluralrules::{PluralRules, PluralRuleType};
-    /// use icu_pluralrules::data::provider::DummyDataProvider;
+    /// use icu_data_provider::InvariantDataProvider;
     ///
     /// let lang: LanguageIdentifier = "en".parse()
     ///     .expect("Failed to parse a language identifier.");
     ///
-    /// let dp = DummyDataProvider::default();
+    /// let dp = InvariantDataProvider;
     ///
     /// let _ = PluralRules::try_new(lang, PluralRuleType::Cardinal, &dp);
     /// ```
     ///
     /// [`type`]: ./enum.PluralRuleType.html
     /// [`data provider`]: ./data/trait.DataProviderType.html
-    pub fn try_new<D: data::provider::DataProviderType>(
-        locale: LanguageIdentifier,
+    pub fn try_new<'d, D: DataProvider<'d>>(
+        langid: LanguageIdentifier,
         type_: PluralRuleType,
         data_provider: &D,
     ) -> Result<Self, PluralRulesError> {
-        let selector = data_provider
-            .get_selector(&locale, type_)?
-            .ok_or(PluralRulesError::MissingData)?;
+        let data_key = match type_ {
+            PluralRuleType::Cardinal => icu_data_key!(plurals: cardinal@1),
+            PluralRuleType::Ordinal => icu_data_key!(plurals: ordinal@1),
+        };
+        let response = data_provider.load(&DataRequest {
+            data_key,
+            data_entry: DataEntry {
```

Comment:
Could we consider renaming it to `key`, and `entry`? I don't think the keys need to indicate that they're data related in a struct called `DataRequest`. @sffc?

Reply:
I'm very open to a discussion on the names of these fields. #244